intermediate_source/torchserve_aws_sagemaker_tutorial.rst

```diff
@@ -0,0 +1,232 @@
+
+Cloning the example repository
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To clone the example repository, enter the following code:
+
+    .. code:: shell
+
+        git clone https://github.com/shashankprasanna/torchserve-examples.git
+        cd torchserve-examples
+
+Clone the TorchServe repository a nd install torch-model-archiver
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Use ``torch-model-archiver`` tool to create a model archive file. The .mar model archive file contains model checkpoints along with it's ``state_dict`` (dictionary object that maps each layer to its parameter tensor).
+
+    .. code:: shell
+
+        git clone https://github.com/pytorch/serve.git
+        pip install serve/model-archiver
+
+To download a PyTorch model and create a TorchServe archive
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+One can customise the handler by passing the `<custom_handler.py> <https://github.com/pytorch/serve/blob/master/docs/custom_service.md>`_ instead of ``image_classifier``
+
+    .. code:: shell
+
+        wget -q https://download.pytorch.org/models/densenet161-8d451a50.pth
+
+        export model_file_name='densenet161'
+        
+        torch-model-archiver --model-name $model_file_name \
+            --version 1.0 --model-file serve/examples/image_classifier/densenet_161/model.py \
+            --serialized-file densenet161-8d451a50.pth \
+            --extra-files serve/examples/image_classifier/index_to_name.json \
+            --handler image_classifier
+
+Uploading the model to Amazon S3
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To upload the model to Amazon S3, complete the following steps:
+
+    #. Create a boto3 session and get the Region and account information
+
+        .. code:: python3 
+
+            import boto3, time, json
+            sess    = boto3.Session()
+            sm      = sess.client('sagemaker')
+            region  = sess.region_name
+            account = boto3.client('sts').get_caller_identity().get('Account')
+
+            import sagemaker
+            role = sagemaker.get_execution_role()
+            sagemaker_session = sagemaker.Session(boto_session=sess)
+
+            Get the default Amazon SageMaker S3 bucket name
+
+            bucket_name = sagemaker_session.default_bucket()
+            prefix = 'torchserve'
+
+    #. Create a compressed tar.gz file out of the densenet161.mar file, because Amazon SageMaker expects models to be in a tar.gz file.
+
+        .. code:: shell
+
+            tar cvfz $model_file_name.tar.gz densenet161.mar
+
+    #. Upload the model to your S3 bucket under the models directory.
+
+        .. code:: shell
+
+            aws s3 cp $model_file_name.tar.gz s3://{bucket_name}/{prefix}/model
+
+Creating an Amazon ECR registry
```

Comment:
This step is needed ONLY if you are going to [BYOD or BYOC](https://docs.aws.amazon.com/sagemaker/latest/dg/your-algorithms-inference-main.html), otherwise it is not needed.